packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts

```diff
@@ -400,16 +411,33 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
         })
       }
 
+      if (
```

Comment:
Would love to see some more comments in here as logic like this is pretty complex and will be hard for other developers to understand what's going on here.